packages/design-system/src/components/contextMenu/menu.js

```diff
@@ -159,11 +177,12 @@ const Menu = ({
         setFocusedId(newSelectedElement?.id || -1);
         return;
       }
+
+      // The direction to move out of a submenu depends on horizontal/vertical and RTL/LTR
+      const keyOut = isHorizontal ? KEYS.UP : isRTL ? KEYS.RIGHT : KEYS.LEFT;
```

Comment:
(Also will depend on if the submenu opens to the top or to the bottom, e.g. if there's no room below.)